# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/libreoffice/hotfix-libcmis.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- libreoffice-24.8.2.1/external/libcmis/0002-hotfix.patch.vanilla	2024-10-11 13:21:02.643531000 +0200
+++ libreoffice-24.8.2.1/external/libcmis/0002-hotfix.patch	2024-10-11 13:21:21.606864818 +0200
@@ -0,0 +1,22 @@
+--- libcmis/src/libcmis/xml-utils.cxx.vanilla	2024-10-02 20:05:57.233290102 +0200
++++ libcmis/src/libcmis/xml-utils.cxx	2024-10-02 20:22:23.179935326 +0200
+@@ -535,7 +535,7 @@
+         boost::uuids::detail::sha1 sha1;
+         sha1.process_bytes( str.c_str(), str.size() );
+ 
+-        unsigned int digest[5];
++        unsigned char digest[20];
+         sha1.get_digest( digest );
+ 
+         stringstream out;
+@@ -543,8 +543,8 @@
+         // hexadecimal digits, including possible leading 0s, or we get
+         // less than 40 digits as result.
+         out << hex << setfill('0') << right;
+-        for ( int i = 0; i < 5; ++i )
+-            out << setw(8) << digest[i];
++        for ( int i = 0; i < sizeof(digest); ++i )
++            out << setw(2) << digest[i];
+         return out.str();
+     }
+ 
--- libreoffice-24.8.2.1/external/libcmis/UnpackedTarball_libcmis.mk.vanilla	2024-10-11 13:21:30.050198393 +0200
+++ libreoffice-24.8.2.1/external/libcmis/UnpackedTarball_libcmis.mk	2024-10-11 13:22:06.993532645 +0200
@@ -25,6 +25,7 @@
 	external/libcmis/0001-cppcheck-operatorEqVarError-in-src-libcmis-http-sess.patch \
 	external/libcmis/0001-Take-into-account-m_CurlInitProtocolsFunction-in-cop.patch \
 	external/libcmis/initprotocols.patch.1 \
+	external/libcmis/0002-hotfix.patch \
 ))
 
 # vim: set noet sw=4 ts=4:
